app/src/hooks/Toasts.tsx

```diff
@@ -1,36 +1,58 @@
-import { toaster } from "@/components/ui/toaster";
+import { Toaster, toaster } from "@/components/ui/toaster";
+import i18next, { TFunction } from "i18next";
+import { useTranslation } from "@/i18n/client";
 
-export function UseSuccessToast(props: {
+export function UseSuccessToast({
+  title,
+  description,
+  duration = 3000,
+  t,
+}: {
   title: string;
-  description: string;
-  text: string;
+  description?: string;
+  duration?: number;
+  t?: TFunction;
 }) {
-  const { title, description, text } = props;
+  const currentLanguage = i18next.language;
+  const { t: tFeedback } = useTranslation(currentLanguage, "feedback");
```

Comment:
Hooks need to be used on the top level of React components usually (this function is called from inside of if statements in a bunch of places though). Let's make sure we're not causing errors because of the hook usages in places where these toasts are emitted and check the changed call sites for errors.